### PR TITLE
Docs Idea: Showing a simple TS example of passing args to a matcher func

### PR DESCRIPTION
**Pull request by msutkowski**
I'm not sure if there is a lot of value here, but we currently don't show this kind of example anywhere and this could potentially save someone time. 

For a more realistic generic use case: https://codesandbox.io/s/rtk-14-addmatcher-counter-test-l11mt?file=/src/features/counter/counterSlice.ts:2037-2492

Let me know what you think.

### CODE REVIEW

File: docs/usage/usage-with-typescript.md [hunk 1]
```diff
@@ -231,6 +231,25 @@ createReducer({ value: 0 }, builder =>
 })
 ```
 
+Additionally, you may want to pass arguments to a matcher function to create reusable helpers. A pattern for that would look like:
+
+```ts
+const isScopedNumberValueAction = (scope: string) => (action: AnyAction): action is PayloadAction<{ value: number }> {
```
Comment (phryneas) [inline docs/usage/usage-with-typescript.md:237]:
```suggestion
const isScopedNumberValueAction = (scope: string) => (action: AnyAction): action is PayloadAction<{ value: number }> => {
```

**Comment by phryneas** [comment]
This could even go a step farther, something like

```ts
const isScopedValueAction = <T>(
    scope: string,
    payloadValidator: (payload: any) => payload is T
  ) => (action: AnyAction): action is PayloadAction<T> => {
    return action.type.startsWith(scope) && payloadValidator(action.payload)
  }

  const isNumber = (v: any): v is number => {
    return typeof v === 'number'
  }

  const isScopedNumberValueAction = (scope: string) =>
    isScopedValueAction(scope, isNumber)

  const slice = createSlice({
    name: 'counter',
    initialState: { value: 0 },
    reducers: {},
    extraReducers: builder => {
      builder.addMatcher(
        isScopedNumberValueAction('counter/'),
        (state, action) => {
          state.value += action.payload
        }
      )
    }
  })
```

But I'm not sure if both of these additions are too much for the "usage with TypeScript" section. Those might be better suited as "recipes".

**Comment by msutkowski** [comment]
@phryneas I actually like showing your full example, to be honest. We know that people like to copy the example code and then do extra 'interesting' implementations with generics that normally result in open issues or discord chatter - and this addresses that. Regarding the recipes, we should probably put together an outline of that what should be included. I've been curating a lot of different ways that things can be typed, but I'm not sure about the best way to organize any of that. The only real issues that consistently pop up seem to be about making `createAsyncThunk`, `createSlice`, or `createEntityAdapter` super generic, so I'm not sure how much would need to be included for recipes.

**Comment by msutkowski** [comment]
Closing this due to #807 potentially being merged and have more examples.

**Pull request closed without merging**